brdnet/evaluate_models.py

```diff
@@ -201,39 +222,49 @@ def eval_tf_model(model_name, lr, train_X, train_Y, val_X, val_Y, checkpoint_dir
 
     seen_pyod_models = set()
 
-    for seed in range(args.num_seeds):
-        # For now, we'll use the load_data function from classifier.py.
-        # In the future, we'll want to tinker with the Z df, so we'll implement a new one
-        train_X, train_Y, val_X, val_Y = classifier.load_data(args.Z_file_path,
-                                                              args.healthy_file_path,
-                                                              args.disease_file_path,
-                                                              seed)
-        val_baseline = utils.get_larger_class_percentage(val_Y)
-
-        for lr in args.learning_rates:
-            for model in model_list:
-                val_acc = None
-                val_auroc = None
-
-                if model_from_tf(model):
-                    val_acc, val_auroc = eval_tf_model(model, lr, train_X, train_Y, val_X, val_Y,
-                                                       args.checkpoint_dir, args.logdir,
-                                                       args.epochs, seed)
-                # Don't rerun pyod models for different learning rates, because they don't have
-                # an lr hyperparameter
-                if (model, seed) not in seen_pyod_models:
-                    if model_from_pyod(model):
-                        val_acc, val_auroc = eval_pyod_model(model, train_X, train_Y, val_X, val_Y)
-                        seen_pyod_models.add((model, seed))
-
-                losses.append((model, lr, seed, val_acc, val_auroc, val_baseline))
-
-    results_df = pandas.DataFrame.from_records(losses, columns=['Model',
-                                                                'LR',
-                                                                'Seed',
-                                                                'val_acc',
-                                                                'val_auroc',
-                                                                'val_baseline',
-                                                               ])
-
-    results_df.to_csv(args.out_path)
+    Z_files = get_Z_files(args.Z_file_dir)
+
+    try:
+        for Z_file_path in Z_files:
```

Comment:
It's the encoding matrix. I forget what it's called in PCA (the loading matrix maybe?), but given the (genes x samples) matrix X, Z is the (genes x latent variables) such that X^T Z gives a lower dimensional representation for the samples. 

I refer to it as the Z matrix throughout the code because the PLIER paper/code refer to it as Z, but I'll give it a more descriptive name if I can think of one.